### PR TITLE
Add support to filter on datasource for active tasks

### PR DESCRIPTION
**Pull request by surekhasaharan**
* Added datasource filter to sql query for active tasks
* Fixed unit tests

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java [hunk 1]
```diff
@@ -632,14 +632,14 @@ public Response getTasks(
...
-      allActiveTaskInfo = taskStorageQueryAdapter.getActiveTaskInfo();
+      allActiveTaskInfo = taskStorageQueryAdapter.getActiveTaskInfo(dataSource);
       for (final TaskInfo<Task> task : allActiveTaskInfo) {
         allActiveTasks.add(
             new AnyTask(
                 task.getId(),
                 task.getTask() == null ? null : task.getTask().getType(),
                 SettableFuture.create(),
-                task.getDataSource(),
+                task.getDataSource(), // ?
```
Comment (gianm) [inline indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java]: What does the `// ?` comment mean?

File: indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java [hunk 2]
```diff
@@ -170,7 +170,7 @@ public void setStatus(TaskStatus status)
   }
 
   @Override
-  public List<TaskInfo<Task>> getActiveTaskInfo()
+  public List<TaskInfo<Task>> getActiveTaskInfo(@Nullable String datasource)
```
[the remark this answers is not in the document]
Comment (surekhasaharan) [reply indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java]: replaced at all the places.

File: indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java (hunk 1 above)
Comment (surekhasaharan) [reply indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java]: removed.

**Comment by gianm** [comment]
LGTM after Travis. thanks @surekhasaharan!

File: indexing-service/src/main/java/io/druid/indexing/overlord/TaskStorage.java [hunk 3]
```diff
@@ -127,17 +127,20 @@
    * Returns a list of currently running or pending tasks as stored in the storage facility as {@link TaskInfo}. No particular order
    * is guaranteed, but implementations are encouraged to return tasks in ascending order of creation.
    *
+   * @param datasource datasource
```
Comment (leventov) [inline indexing-service/src/main/java/io/druid/indexing/overlord/TaskStorage.java]: This line broke TeamCity CI: https://teamcity.jetbrains.com/viewLog.html?buildId=1531809&tab=Inspection&buildTypeId=OpenSourceProjects_Druid_Inspections

**Pull request merged**